rest_framework_json_api/serializers.py

```diff
@@ -149,8 +150,55 @@ def validate_path(serializer_class, field_path, path):
         super(IncludedResourcesValidationMixin, self).__init__(*args, **kwargs)
 
 
+class LazySerializersDict(MutableMapping):
+    def __init__(self, klass, serializers):
+        self.klass = klass
+        self.serializers = serializers
+
+    def __getitem__(self, key):
+        value = self.serializers[key]
+        if not isinstance(value, type):
+            if value == "self":
+                value = self.klass
+
+            value = import_class_from_dotted_path(value)
+            self.serializers[key] = value
+
+        return value
+
+    def __setitem__(self, key, field):
+        self.serializers[key] = field
+
+    def __delitem__(self, key):
+        del self.serializers[key]
+
+    def __iter__(self):
+        return iter(self.serializers)
+
+    def __len__(self):
+        return len(self.serializers)
+
+    def __repr__(self):
+        return dict.__repr__(self.serializers)
+
+
 class SerializerMetaclass(SerializerMetaclass):
-    pass
+    def __new__(cls, name, bases, attrs):
+        serializer_class_path = attrs["__module__"] + "." + name
```

Comment:
I'm not happy about this as it feels really hacky and can go wrong in many ways.
I wrote it just to not make a regression.
Django allows this, but one need proper understanding of Django to understand how this works.

Reply:
What is the reason to use a lazy dictionary? This will only be done once during start up and a fairly simple task. And this would solve this issue that you do not need this workaround right?

Reply:
This is only for `"self"` in included_serializers, as the serializer is yet to be initialized, and we have to somehow have a reference to it, so I use this hacky way to get the module path of the serializer, to be called later lazily...

Reply:
I just think this lazy loading is way to overkill for a simple task like this. So am I understanding you correct you only do this lazy loading because of `self` reference?

I have a feeling there needs to be a way not to do this lazily.

Reply:
As you see, we are currently in the meta class of the class, meaning the class itself doesn't yet exist.
I don't see any other option but to lazy load it later somehow.
It seems that Django saves `to="self"` till it actually adds the field to the model.

Reply:
Sure here it would not work, but I was rather thinking whether there is a better spot for this code snippet (e.g. in the view). It seems code in another spot might get even more complex though.

Anyway this lazy dict in my opinion does not need to be a `MutableMapping` though, a `Mapping` is enough which makes the code less verbose as well (a few methods can be deleted).

To solve the `self` reference issue another idea I had is that super needs to be called first and the klass can be passed on to `LazySerializersDict`.

```python
    def __new__(cls, name, bases, attrs):
        klass = super().__new__(cls, name, bases, attrs)
        # assign lazy dict here where klass can be passed on for self
        return klass
``` 

Just on the top of my head not sure I have overlooked anything but I guess worth a try.